src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterface.cs

```diff
@@ -174,6 +180,38 @@ public override string ReadLine()
             return ReadLine(false, string.Empty, out unused, true, true);
         }
 
+        /// <summary>
+        /// See base class
+        /// </summary>
+        /// <returns></returns>
+        /// <exception cref="HostException">
+        /// If obtaining a handle to the active screen buffer failed
+        ///    OR
+        ///    Win32's setting input buffer mode to disregard window and mouse input failed
+        ///    OR
+        ///    Win32's ReadConsole failed
+        /// </exception>
+        /// <exception cref="PipelineStoppedException">
+        /// If Ctrl-C is entered by user
+        /// </exception>
+        public override string ReadLineMaskedAsString()
+        {
+            HandleThrowOnReadAndPrompt();
+
+            // we lock here so that multiple threads won't interleave the various reads and writes here.
+
+            object result = null;
+            lock (_instanceLock)
+            {
+                result = ReadLineSafe(false, PrintToken);
+            }
+
+            StringBuilder resultSb = result as StringBuilder;
```

Comment:
I wonder why do we return "object", not typed result from ReadLineSafe()?

Reply:
Thanks for the comments and for taking the mark all the styling errors :)

My understanding is that this a poor man's version of template programming for the following 2 cases:

1. If I need to return a `secureString` then I have to build it from the start. Using a `StringBuilder` here would defeat the purpose of a `secureString`.
2. If I need to return a `string` then I can use a `StringBuilder` to be more efficient. I guess you could use a string here but that would involve a lot of object allocations.

I did go through the documentation and I toyed a bit with the idea but I could not think of an elegant way to make this function return a type. Do you have any suggestion?

Reply:
Thanks for clarify! ReadLineSafe() method has so many code paths that it makes no sense change anything in the PR.